pkg/northbound/gnmi/set.go

```diff
@@ -208,10 +221,41 @@ func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
 		}
 		networkChanges[store.ConfigName(configName)] = changeID
 	}
-
 	return updateResults, networkChanges, nil
 }
 
+func listenForDeviceResponse(changes mapNetworkChanges, target string, name store.ConfigName) (bool, error) {
+	mgr := manager.GetManager()
+	respChan, ok := mgr.Dispatcher.GetResponseListener(topocache.ID(target))
+	if !ok {
+		log.Infof("Device %s not properly registered, not waiting for southbound confirmation", target)
+		return true, nil
+	}
+	response := <-respChan
+	switch eventType := response.EventType(); eventType {
+	case events.EventTypeAchievedSetConfig:
+		return true, nil
+	case events.EventTypeErrorSetConfig:
+		//Removing previously applied Previously applied
+		for t := range changes {
+			err := mgr.ConfigStore.RemoveLastChangeEntry(t)
+			if err != nil {
+				log.Error("Can't remove last entry for ", t, err)
+			}
+			//TODO calculate the reverse and send down.
+		}
+		//Removing the failed target
+		err := mgr.ConfigStore.RemoveLastChangeEntry(name)
```

Comment:
should not be necessary to rollback on current device, I thought it would not be added until you had a success

Reply:
it actually is, line 103 of setconfig.go stores it in the configuration.

Reply:
Maybe needs to hold off, but that's another days work

Reply:
yes, I'd merge it as is. Let me put a TODO in there